src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/NuGetCredentials.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Credentials;
+using NuGet.Protocol;
+using NuGet.Protocol.Plugins;
+
+namespace Microsoft.DotNet.UpgradeAssistant.MSBuild
+{
+    public static class NuGetCredentials
```

Comment:
could this be something done via a factory pattern through DI?

Reply:
`HttpHandlerResourceV3.CredentialService` is, unfortunately, static so it seems like the simplest way to set it is with a static method. What would a DI factory pattern look like here?

Reply:
After looking at the other comments I'm thinking it would be better to have this come from DI. I think I'll make a few small changes to turn it into an `IUpgradeStartup`.